packages/dds/tree/src/tree/treeTextFormat.ts

```diff
@@ -160,6 +160,7 @@ export function setGenericTreeField<T>(
 export function genericTreeKeys<T>(tree: GenericFieldsNode<T>): readonly FieldKey[] {
```

Comment:
I believe most if not all uses of JsonableTree users already ensure empty fields never occur in their trees. Rather than slowing down the tree access with filtering (which I suspect is a significant performance regression, but if not, let me know and maybe use your approach), I suggest just documenting that this includes all fields on the objects, and noe that call user of this function might need to skip over empty fields if their trees include them and they need to omit them.

Reply:
If we do want to filter out empty fields (instead of just not include them in the first place) we risk memory leaks from unbounded numbers of empty fields showing up in map style nodes and deep compares of jsonableTrees won't work for equality (some of our tests depend on this).

Also, if we do want to keep the filter approach anyway, I suspect its cheaper to do the filtering in the implementation of first/next field, since it saves a copy and is more lazy.

Reply:
Since there are a lot of problems with the filter approach, I decided to skip fields in first/nextField instead.